testnet/stacks-node/src/nakamoto_node/signer_coordinator.rs

```diff
@@ -336,10 +358,29 @@ impl SignerCoordinator {
                         return Err(NakamotoNodeError::BurnchainTipChanged);
                     }
 
+                    if rejections_timer.elapsed() > rejections_timeout {
+                        return Err(NakamotoNodeError::SigningCoordinatorFailure(
+                            "Gave up while tried reaching the threshold".into(),
+                        ));
```

Comment:
I think that it would be good to add a `warn!` log here (and the other check location), with a little more information about the timeout:

```rust
warn!("Timed out while waiting for responses from signers";
            "elapsed" => rejections_timer.elapsed().as_secs(),
            "rejections_timeout" => rejections_timeout.as_secs(),
            "rejections" => rejections,
            "rejections_threshold" => self.total_weight.saturating_sub(self.weight_threshold)
  );
  ```